app/controllers/audio_files_controller.rb

```diff
@@ -6,12 +6,31 @@ def new
   def create
     @audio_file = current_user.audio_files.new(audio_file_params)
     if @audio_file.save
-      redirect_to dashboard_url, notice: "Audio file saved successfully"
+      Activity.create!(
+        subject: @audio_file,
+        user: current_user,
+      )
+      redirect_to dashboard_url, notice: "Audio file created! You rock."
     else
       render :new
     end
   end
 
+  def edit
+    @audio_file = current_user.audio_files.find(params[:id])
+  end
+
+  def update
+    @audio_file = current_user.audio_files.find(params[:id])
+    if @audio_file.update(audio_file_params)
+      Activity.create!(
```

Comment:
This may be for a subsequent PR, are you thinking of differentiating the activity types based on editing / creating?